### PR TITLE
Count to expiry, total number SANS

### PR DESCRIPTION
**Pull request by McJoppy**

### CODE REVIEW

File: files/graphite-scripts/letsencrypt-metrics [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/php
```
Comment (halkyon) [inline files/graphite-scripts/letsencrypt-metrics]: `#!/usr/bin/env php` would be more portable

File: files/graphite-scripts/letsencrypt-metrics [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+require_once '/opt/graphite/graphite_functions.php';
+
+list($serverName,) = explode(".", trim(`hostname -f`));
+
+$now = time();
+$numberSANS = 0;
+
+foreach (glob("/etc/nginx/letsencrypt/*crt") as $certFile) {
```
Comment (halkyon) [inline files/graphite-scripts/letsencrypt-metrics]: was going to ask if this path could be changed, i.e. so we could change the path if we decided to store the certs elsewhere for SSP... but it doesn't look like there's any way to set the parameters for individual metrics collector scripts in this module

File: files/graphite-scripts/letsencrypt-metrics [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    $certName = basename($certFile);
+    $certName = substr($certName, 0, strrpos($certName, '.'));
+    $certData = openssl_x509_parse(file_get_contents($certFile));
+    $certExpiry = $certData['validTo_time_t'];
+
+    if (isset($certData['extensions'])) {
+        $certSANS = $certData['extensions']['subjectAltName'];
+        $numberSANS ++;
```
Comment (halkyon) [inline files/graphite-scripts/letsencrypt-metrics]: minor: convention has it as `++$numberSANS`

File: files/graphite-scripts/letsencrypt-metrics (hunk 2 above)
Comment (halkyon) [reply files/graphite-scripts/letsencrypt-metrics]: or maybe there is, we could make this a puppet template and parameterise the cert path?

Comment (McJoppy) [reply files/graphite-scripts/letsencrypt-metrics]: The naming of metrics etc make it relevant to our Let's Encrypt tools.

The path is hard coded currently, to `/etc/nginx/letsencrypt/` in the puppet install.

**Pull request merged**